File: frontend/src/lib/dto/container-registry-dto.ts
export interface ContainerRegistryCreateDto {
	url: string;
	username: string;
	token: string;
	description?: string;
	insecure?: boolean;
	enabled?: boolean;
}

export interface ContainerRegistryUpdateDto {
	url?: string;
	username?: string;
	token?: string;
	description?: string;
	insecure?: boolean;
	enabled?: boolean;
}

File: backend/internal/dto/container_registry_dto.go
package dto

import "time"

type ContainerRegistryDto struct {
	ID          string    `json:"id"`
	URL         string    `json:"url"`
	Username    string    `json:"username"`
	Description *string   `json:"description,omitempty"`
	Insecure    bool      `json:"insecure"`
	Enabled     bool      `json:"enabled"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

type ContainerRegistryCredential struct {
	URL      string `json:"url" binding:"required"`
	Username string `json:"username" binding:"required"`
	Token    string `json:"token" binding:"required"`
	Enabled  bool   `json:"enabled"`
}
